src/ray/common/client_connection.cc

```diff
@@ -28,8 +28,46 @@
 #include "ray/common/ray_config.h"
 #include "ray/util/util.h"
 
+#if defined(_WIN32)
+#include <Windows.h>
+#else
+#include <fcntl.h>
+#include <unistd.h>
+#endif
+
 namespace ray {
 
+namespace {
+
+#if defined(_WIN32)
+
+// Don't care what exact type is in windows... Looks like to be an asio specific type.
+template <typename NativeHandleType>
+void setFdCloseOnFork(const NativeHandleType &handle) {
+  // In Windows we don't need to do anything, beacuse in CreateProcess we pass
+  // bInheritHandles = false which means we don't inherit handles or sockets.
+  return;
+}
+#else
+void setFdCloseOnFork(int fd) {
```

Comment:
what's happening if we call it on the same fd twice? No-op?

Reply:
idempotent. still sets to close-on-fork.